### PR TITLE
update keyvault api version to 2019-09-01

### PR DESCRIPTION
**Pull request by rolandmkunkel**
### Which issue this PR addresses:

Updates the keyvault API version to 2019-09-01

### What this PR does / why we need it:

The newer API version contains models for key, which allows us to create keys through templates.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -17,6 +17,7 @@ require (
 	github.com/apparentlymart/go-cidr v1.1.0
 	github.com/aws/aws-sdk-go v1.38.46 // indirect
 	github.com/axw/gocov v1.0.0
+	github.com/azure/azure-sdk-for-go v56.2.0+incompatible // indirect
```
Comment (m1kola) [inline go.mod:20]: It semes like conflicts were resolved incorrectly. Try to run `go mod tidy` and `go mod vendor`: it removes changes from `go.mod`, `go.sum` and `vendor/modules.tx`.

**Comment by m1kola** [comment]
@rolandmkunkel I see the updated version of the PR. What version of Go you use? When I ran these commands with 1.14.15 against the previous version of the PR it only removed `github.com/azure/azure-sdk-for-go` related stuff from `go.sum`.

**Review by troy0820 — CHANGES_REQUESTED** [review]
nit: I tend to make any changing of dependencies `vendor` a separate commit.  This will help with what @m1kola is seeing as well.  I did a go mod vendor and it added everything back that was deleted in `go.sum` and I'm using go 1.15.14.  

Then I would make the second commit the make generate/changing the apiversions in `pkg/util/azureclient/apiversions.go`
- [ ] Make dependencies vendor a separate commit
- [ ] Create commit with changes to `apiversions.go` and `make generate`

**Comment by rolandmkunkel** [comment]
I didn't have to update my SDK, so all is in one commit now

**Review by m1kola — APPROVED** [review]

**Review by troy0820 — APPROVED** [review]

**Pull request merged**